### ISSUE TITLE
Childprocess 5.0 gets conflict with ScreenRecorder now

### ISSUE THREAD

**Issue by Rajagopalan-M** [kapoorlakshya/screen-recorder#105]
Screen recorder works with ChildProcess 4.1.0 but it's not working with 5.0. Can you increase this limit?

last time same issue happened and you fixed here https://github.com/kapoorlakshya/screen-recorder/issues/99

```
C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1480:in `rescue in block in activate_dependencies': Could not find 'childprocess' (>= 1.0, < 5.0) among 114 total gem(s) (Gem::MissingSpecError)
Checked in 'GEM_PATH=C:/Users/rajagopalan.m/.local/share/gem/ruby/3.3.0;C:/Ruby/lib/ruby/gems/3.3.0' at: C:/Ruby/lib/ruby/gems/3.3.0/specifications/screen-recorder-1.6.0.gemspec, execute `gem env` for more information
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1477:in `block in activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `each'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1448:in `activate'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems.rb:205:in `rescue in try_activate'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems.rb:198:in `try_activate'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `block in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `synchronize'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `rescue in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:135:in `require'
        from C:/A/TestBot/TestBot.rb:20:in `map'
        from C:/A/TestBot/TestBot.rb:20:in `initialize'
        from C:/A/TestBot/Run.rb:6:in `new'
        from C:/A/TestBot/Run.rb:6:in `<main>'
C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/dependency.rb:315:in `to_specs': Could not find 'childprocess' (>= 1.0, < 5.0) - did find: [childprocess-5.0.0] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=C:/Users/rajagopalan.m/.local/share/gem/ruby/3.3.0;C:/Ruby/lib/ruby/gems/3.3.0' , execute `gem env` for more information
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1478:in `block in activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `each'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1448:in `activate'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems.rb:205:in `rescue in try_activate'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems.rb:198:in `try_activate'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `block in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `synchronize'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `rescue in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:135:in `require'
        from C:/A/TestBot/TestBot.rb:20:in `map'
        from C:/A/TestBot/TestBot.rb:20:in `initialize'
        from C:/A/TestBot/Run.rb:6:in `new'
        from C:/A/TestBot/Run.rb:6:in `<main>'
C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1480:in `rescue in block in activate_dependencies': Could not find 'childprocess' (>= 1.0, < 5.0) among 114 total gem(s) (Gem::MissingSpecError)
Checked in 'GEM_PATH=C:/Users/rajagopalan.m/.local/share/gem/ruby/3.3.0;C:/Ruby/lib/ruby/gems/3.3.0' at: C:/Ruby/lib/ruby/gems/3.3.0/specifications/screen-recorder-1.6.0.gemspec, execute `gem env` for more information
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1477:in `block in activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `each'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1448:in `activate'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems.rb:199:in `try_activate'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `block in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `synchronize'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `rescue in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:135:in `require'
        from C:/A/TestBot/TestBot.rb:20:in `map'
        from C:/A/TestBot/TestBot.rb:20:in `initialize'
        from C:/A/TestBot/Run.rb:6:in `new'
        from C:/A/TestBot/Run.rb:6:in `<main>'
C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/dependency.rb:315:in `to_specs': Could not find 'childprocess' (>= 1.0, < 5.0) - did find: [childprocess-5.0.0] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=C:/Users/rajagopalan.m/.local/share/gem/ruby/3.3.0;C:/Ruby/lib/ruby/gems/3.3.0' , execute `gem env` for more information
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1478:in `block in activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `each'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1466:in `activate_dependencies'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/specification.rb:1448:in `activate'
        from C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems.rb:199:in `try_activate'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `block in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `synchronize'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:139:in `rescue in require'
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:135:in `require'
        from C:/A/TestBot/TestBot.rb:20:in `map'
        from C:/A/TestBot/TestBot.rb:20:in `initialize'
        from C:/A/TestBot/Run.rb:6:in `new'
        from C:/A/TestBot/Run.rb:6:in `<main>'
<internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require': cannot load such file -- screen-recorder (LoadError)
        from <internal:C:/Ruby/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
        from C:/A/TestBot/TestBot.rb:20:in `map'
        from C:/A/TestBot/TestBot.rb:20:in `initialize'
        from C:/A/TestBot/Run.rb:6:in `new'
        from C:/A/TestBot/Run.rb:6:in `<main>'

Process finished with exit code 1

```

**Comment by kapoorlakshya** [comment]
@Rajagopalan-M I'll try to get a release out in a week or so.

**Comment by Rajagopalan-M** [comment]
Sure thanks.

**Comment by kapoorlakshya** [comment]
Also which version of watir/selenium webdriver are you using? Only childprocess < 5.0 is supported by selenium-webdriver [up to v4.6](https://rubygems.org/gems/selenium-webdriver/versions/4.6.0) and then removed in selenium-webdriver [v4.7 onward](https://rubygems.org/gems/selenium-webdriver/versions/4.7.0). I am also considering removing childprocess in favor of using `Process.spawn` - https://github.com/SeleniumHQ/selenium/issues/11251.

**Comment by kapoorlakshya** [comment]
@Rajagopalan-M Give this branch a try:

```rb
gem 'screen-recorder', git: 'https://github.com/kapoorlakshya/screen-recorder.git', branch: 'update_deps'
```

it removes `childprocess` and `os` gems so you can use whatever version you like for these.

**Comment by Rajagopalan-M** [comment]
> @Rajagopalan-M Give this branch a try:
> 
> ```ruby
> gem 'screen-recorder', git: 'https://github.com/kapoorlakshya/screen-recorder.git', branch: 'update_deps'
> ```
> 
> it removes `childprocess` and `os` gems so you can use whatever version you like for these.

I don't know how to use this. I always install using `gem install screen-recorder` and then I will check.

**Comment by kapoorlakshya** [comment]
> > @Rajagopalan-M Give this branch a try:
> > 
> > ```ruby
> > gem 'screen-recorder', git: 'https://github.com/kapoorlakshya/screen-recorder.git', branch: 'update_deps'. 
> > ```
> > 
> > it removes `childprocess` and `os` gems so you can use whatever version you like for these.
> 
> I don't know how to use this. I always install using `gem install screen-recorder` and then I will check. Or can you commit to the main branch and let me install the new version of screen recorder gem and check it?

This is to be placed in a Gemfile and then installed using bundler - https://bundler.io/guides/gemfile.html

It's the recommended way for managing gems in a project.

**Comment by Rajagopalan-M** [comment]
> > > @Rajagopalan-M Give this branch a try:
> > > ```ruby
> > > gem 'screen-recorder', git: 'https://github.com/kapoorlakshya/screen-recorder.git', branch: 'update_deps'. 
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > it removes `childprocess` and `os` gems so you can use whatever version you like for these.
> > 
> > 
> > I don't know how to use this. I always install using `gem install screen-recorder` and then I will check. Or can you commit to the main branch and let me install the new version of screen recorder gem and check it?
> 
> This is to be placed in a Gemfile and then installed using bundler - https://bundler.io/guides/gemfile.html
> 
> It's the recommended way for managing gems in a project.

I don't use Gem file in my project and so I don't use bundler either.

**Comment by kapoorlakshya** [comment]
Closed by #107. v1.7.0 has been released.

**Comment by Rajagopalan-M** [comment]
> Closed by #107. v1.7.0 has been released.

Thanks, I will check and let you know.
